### PR TITLE
fix(create-medusa-app): improved error messages

### PR DESCRIPTION
**Pull request by shahednasser**
- Fixed error message for PostgreSQL that previously showed `EOL` in the output message instead of actually outputting a new line
- Modified the docs URLshown  in the PostgreSQL error message to toggle the os tab based on the user's OS in the docs
- Fixed uncaught error when the process is terminated before a database client is defined

### CODE REVIEW

File: packages/create-medusa-app/src/utils/get-current-os.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const getCurrentOs = (): string => {
+  switch (process.platform) {
```
Comment (srindom) [inline packages/create-medusa-app/src/utils/get-current-os.ts:2]: question: is this sufficient?

File: packages/create-medusa-app/src/utils/create-db.ts [hunk 2]
```diff
@@ -80,9 +82,7 @@ export async function getDbClientAndCredentials(dbName: string): Promise<{
       })
     } catch (e) {
       logMessage({
-        message:
-          "Couldn't connect to PostgreSQL. Make sure you have PostgreSQL installed and the credentials you provided are correct.${EOL}${EOL}" +
-          "You can learn how to install PostgreSQL here: https://docs.medusajs.com/development/backend/prepare-environment#postgresql",
+        message: `Couldn't connect to PostgreSQL. Make sure you have PostgreSQL installed and the credentials you provided are correct.${EOL}${EOL}You can learn how to install PostgreSQL here: https://docs.medusajs.com/development/backend/prepare-environment?os=${getCurrentOs()}#postgresql`,
```
Comment (srindom) [inline packages/create-medusa-app/src/utils/create-db.ts:85]: praise: cool idea with the os query param 🙌

File: packages/create-medusa-app/src/utils/get-current-os.ts (hunk 1 above)
Comment (shahednasser) [reply packages/create-medusa-app/src/utils/get-current-os.ts:2]: Should be 😄 According to the Node.js docs it's the equivalent to [os.platform](https://nodejs.org/docs/latest-v16.x/api/os.html#osplatform) method. We can also ask someone with a different OS than mac to test it out.

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**